src/components/ProjectSettingsTabs/CreateProjectModal.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from "react";
+import { useMutation } from "@apollo/client";
+import styled from "@emotion/styled";
+import { Size, Variant } from "@leafygreen-ui/button";
+import { Button } from "components/Button";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import { SpruceForm, SpruceFormProps } from "components/SpruceForm";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  CreateProjectMutation,
+  CreateProjectMutationVariables,
+} from "gql/generated/types";
+import { CREATE_PROJECT } from "gql/mutations";
+
+const getModalFormDefinition = (owner: string, repo: string) => ({
+  schema: {
+    type: "object" as "object",
+    properties: {
+      projectName: {
+        type: "string" as "string",
+        title: "Project Name",
+      },
+      projectId: {
+        type: "string" as "string",
+        title: "Project Id",
+        value: "hi",
+      },
+      owner: {
+        type: "string" as "string",
+        title: "Owner",
+        default: owner,
+      },
+      repo: {
+        type: "string" as "string",
+        title: "Repo",
+        default: repo,
+      },
+    },
+  },
+  uiSchema: {
+    projectName: {
+      "ui:data-cy": "project-name-input",
+      "ui:showErrors": false,
+    },
+    projectId: {
+      "ui:data-cy": "project-id-input",
+      "ui:description":
+        "Optionally enter an immutable project ID that would be used by Evergreen internally instead of defaulting to a random hash; An id should only be user-specified with good reason, such as if the project will be using performance tooling. \n It cannot be changed!",
+    },
+    owner: {
+      "ui:data-cy": "owner-input",
+      "ui:showErrors": false,
+    },
+    repo: {
+      "ui:data-cy": "repo-input",
+      "ui:showErrors": false,
+    },
+  },
+});
+interface Props {
+  owner: string;
+  repo: string;
+}
+export const CreateProjectModal: React.FC<Props> = ({ owner, repo }) => {
+  const dispatchToast = useToastContext();
+  const modalFormDefinition = getModalFormDefinition(owner, repo);
+  const isAdmin = true; // todo after EVG-16353
+  const [open, setOpen] = useState(false);
+
+  const onCancel = () => setOpen(false);
+  const onConfirm = (formState) => {
+    createProject({
+      variables: {
+        project: {
+          identifier: formState.projectName,
+          id: formState.projectId,
+          owner: formState.owner,
+          repo: formState.repo,
+        },
+      },
+    });
+    setOpen(false);
+  };
+  const [formState, setFormState] = useState({});
+  const [hasError, setHasError] = useState(false);
+  const isDisabled =
+    Object.keys(formState).length === 0 ||
+    Object.values(formState).some((input) => input === "") ||
+    hasError;
```

Comment:
I recently made some improvements to validation on a modal that was using this approach; [here](https://github.com/evergreen-ci/spruce/pull/1095/commits/4ea430dcdc4b170c4ad9fec8098e4060e79d99cf#diff-1ac4434cc610fed45078b29eaafe1d252602283a4dc51a049f157bd80693c7c6) is the commit where that happened. I think your intuition was correct that avoiding a validation function is preferable, so hopefully using a combination of `minLength` etc. rules + [RJSF dependencies](https://react-jsonschema-form.readthedocs.io/en/latest/usage/dependencies/) can mean a function is not required!

Reply:
fantastic!! added.